File: guides/micronaut-graphql-todo/kotlin/src/main/kotlin/example/micronaut/Author.kt
package example.micronaut

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.GeneratedValue.Type.AUTO
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import jakarta.validation.constraints.NotNull

@MappedEntity // <1>
class Author(val username: @NotNull String?) {

    @Id // <2>
    @GeneratedValue(AUTO)
    var id: Long? = null
}

File: guides/micronaut-graphql-todo/python/src/example/micronaut/author.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotNull
from micronaut.core.annotation import Introspected
from micronaut.data.annotation import GeneratedValue, Id, MappedEntity


@dataclass
@Introspected
@MappedEntity  # <1>
class Author:
    username: Annotated[str, NotNull]
    id: Annotated[int | None, Id, GeneratedValue] = None  # <2>
